sklearn/tree/_criterion.pyx

```diff
@@ -1325,3 +1327,122 @@ cdef class FriedmanMSE(MSE):
 
         return (diff * diff / (self.weighted_n_left * self.weighted_n_right *
                                self.weighted_n_node_samples))
+
+
+cdef class Poisson(RegressionCriterion):
+    """Half Poisson deviance as impurity criterion.
+
+    Poisson deviance = 2/n * sum(y_true * log(y_true/y_pred) + y_pred - y_true)
+
+    Note that the deviance is >= 0, and for trees with `y_pred = mean(y_true)`,
+    one always has `sum(y_pred - y_true) = 0`. It remains the implemented
+    impurity:
+        1/n * sum(y_i * log(y_i/y_pred)
+    """
+    # FIXME in 0.25:
```

Comment:
Is this relevant considering we have `proxy_impurity_improvement`?

Reply:
Isn't the impurity calculated for each node (after a split)? If so, the FIXME might improve performance as `xlogy` must be calculated for every sample in the node. There is no algebraic simplification.
I have not done benchmakrs though.